Cannot go to implementations of classes in external packages

**Describe the bug**

In a file of my project, I import a class from an external package by

```scala
import com.amazonaws.services.s3.AmazonS3Client
```

The package is hosted in a custom host and I need to connect to a VPN to download it. I can go to the implementations of `AmazonS3Client`, which looks like

```java
public class AmazonS3Client extends AmazonWebServiceClient implements AmazonS3 {
...
```

Then I want to go to the implementations of `AmazonS3` but it shows errors

```
[Error - 6:56:31 PM] Request textDocument/codeLens failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.nio.file.FileSystemNotFoundException: /Users/me/Library/Caches/Coursier/v1/https/maven.localproxy.net:42042/repo/com/amazonaws/aws-java-sdk-s3/1.9.6/aws-java-sdk-s3-1.9.6-sources.jar
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1(CancelTokens.scala:40)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1$adapted(CancelTokens.scala:38)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:484)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.nio.file.FileSystemNotFoundException: /Users/me/Library/Caches/Coursier/v1/https/maven.localproxy.net:42042/repo/com/amazonaws/aws-java-sdk-s3/1.9.6/aws-java-sdk-s3-1.9.6-sources.jar
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystem.<init>(ZipFileSystem.java:104)
	at jdk.zipfs/jdk.nio.zipfs.ZipFileSystemProvider.newFileSystem(ZipFileSystemProvider.java:106)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:344)
	at java.base/java.nio.file.FileSystems.newFileSystem(FileSystems.java:293)
	at scala.meta.internal.io.PlatformFileIO$.newFileSystem(PlatformFileIO.scala:104)
	at scala.meta.internal.mtags.MtagsEnrichments$XtensionURIMtags.toAbsolutePath(MtagsEnrichments.scala:110)
	at scala.meta.internal.mtags.MtagsEnrichments$XtensionStringMtags.toAbsolutePath(MtagsEnrichments.scala:137)
	at scala.meta.internal.mtags.MtagsEnrichments$XtensionStringMtags.toAbsolutePath(MtagsEnrichments.scala:135)
	at scala.meta.internal.metals.MetalsEnrichments$XtensionString.toAbsolutePath(MetalsEnrichments.scala:623)
	at scala.meta.internal.metals.MetalsEnrichments$XtensionString.toAbsolutePath(MetalsEnrichments.scala:620)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$codeLens$2(MetalsLanguageServer.scala:1689)
	at scala.meta.internal.metals.TimerProvider.timedThunk(TimerProvider.scala:25)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$codeLens$1(MetalsLanguageServer.scala:1688)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:678)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:467)
	... 3 more
```

In my local, the path of the file is

```
/Users/me/Library/Caches/Coursier/v1/https/maven.localproxy.net%3A42042/repo/com/amazonaws/aws-java-sdk-s3/1.9.6/aws-java-sdk-s3-1.9.6-sources.jar
```

**Expected behavior**

I should be able to go to the implementations of `AmazonS3`.

**Installation:**

- Operating system: macOS
- VSCode version: 1.71.0
- VSCode extension version: v1.19.9
- Metals version: 0.11.8+79-e6494d3f-SNAPSHOT

--- a/mtags/src/main/scala-2/scala/meta/internal/mtags/MtagsEnrichments.scala
+++ b/mtags/src/main/scala-2/scala/meta/internal/mtags/MtagsEnrichments.scala
@@ -136,7 +136,8 @@ trait MtagsEnrichments extends CommonMtagsEnrichments {
       else if (value.toUpperCase.startsWith("JAR:FILE%3A"))
         URLDecoder.decode(value, "UTF-8").toAbsolutePath(followSymlink)
       else if (value.toUpperCase.startsWith("JAR"))
-        URI.create(value).toAbsolutePath(followSymlink)
+        new URI("jar", value.stripPrefix("jar:"), null)
+          .toAbsolutePath(followSymlink)
       else {
         val stripped = value.stripPrefix("metals:")
         val percentEncoded = URIEncoderDecoder.encode(stripped)

